src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java

```diff
@@ -94,6 +100,9 @@
   @RequestMapping(value = "/{type}/fileupload", method = RequestMethod.POST, consumes = "multipart/form-data")
   public final @ResponseBody ResponseEntity<SampleSummary> uploadSampleFile(@PathVariable("type") final String type, @RequestParam("file") MultipartFile file) throws CTPException {
     log.debug("Entering Sample file upload for Type {}", type);
+    if (!Arrays.asList("B", "census", "social").contains(type)) {
+      return ResponseEntity.badRequest().build();
+    }
```

Comment:
Could we not just catch the UnsupportedOperationException when making the sampleService.ingest call instead of duplicating types?